### PR TITLE
mega commit for user persistence and orders

### PR DESCRIPTION
**Pull request by kozr**
users are created within each session
got rid of required email
user can log in
user can share link
user can register for a lobby/ session
user can find their old orders
user can make orders

todo: update subtotal stuff
todo: just get rid of confirm order button? (because right now I update every time there's a change  anyway)
todo: consolidate page

### CODE REVIEW

File: server/validation/register.js [hunk 1]
```diff
@@ -4,23 +4,13 @@ module.exports = function validateRegisterInput(data) {
...
-    errors.email = "Email field is required";
-  } else if (!Validator.isEmail(data.email)) {
-    errors.email = "Email is invalid";
-  }
   // Password checks
-  if (Validator.isEmpty(data.password)) {
-    errors.password = "Password field is required";
-  }
-  if (!Validator.isLength(data.password, { min: 6, max: 30 })) {
+  if (data.password && !Validator.isLength(data.password, { min: 6, max: 30 })) {
```
Comment (acchiang) [inline server/validation/register.js]: have you been able to get this error to show up? I haven't, and I get a different error but the user creation still goes through 
Screen Shot 2021-08-09 at 4 47 31 PM
image

File: server/controllers/session-ctrl.js [hunk 2]
```diff
@@ -78,36 +83,70 @@ updateSessionName = async (req, res) => {
...
+      );
+    } else {
+      res.status(400).json({ passwordincorrect: "Password incorrect or username is taken" });
     }
-  );
-};
+  }
+}
 
+// i hate this method because I couldn't find a clean way to do it, could use help
```
Comment (acchiang) [inline server/controllers/session-ctrl.js]: let's just remove this comment lol

**Review by acchiang — APPROVED** [review]
you're right, mega commit

File: server/validation/register.js (hunk 1 above)
Comment (kozr) [reply server/validation/register.js]: oh lol I forgot this existed, I just used my own validation instead (ofc it had bugs)

Comment (kozr) [reply server/validation/register.js]: there's client side validationa nd the server side one. The one you saw is the half implemented client one.

**Pull request merged**